design/authenticators-status/authn_status_oidc.md

```diff
@@ -0,0 +1,74 @@
+# Feature Overview
+
+***As a*** Conjur operator\
+***I'd like to*** know that the OIDC authenticator is configured correctly\
+***So that*** I can complete the configuration properly if it's incomplete (or remove
+it if it's not needed).
+
+This feature lets the person who configures the OIDC authenticator to get an immediate feedback
+on the configuration, before any user needs to run an authentication request.
+
+As mentioned in the general status check implementation details, in addition to the general validations,
+some of the authenticators will have additional validations based on their specific requirements.  
+
+***Note***: At this time, we will only be implementing the OIDC status check. 
+Later implementations of authenticator status checks will be done incrementally. 
+Therefore, if the authenticator does not have the status check implemented, then a 503 error will 
+be returned. 
+
+# Implementation Details
+
+As mentioned in the general status check implementation details, we will add the following for the 
+OIDC status check:
+
+1. A `Status` CommandClass in the structure `Authentication::AuthnOidc::Status`
+1. A `status` method to its existing authenticator class (i.e. `Authentication::AuthnOidc::Authenticator`)
+which will call the new CommandClass, as follows: 
+ 
+ ```
+ def status
+   Authentication::AuthnOidc::Status.new.(
+     <input for status check>
+   )
+ end
+ ``` 
+
+The new CommandClass `Authentication::AuthnOidc::Status` will consist of a
+`call` method that will perform the following checks:
+
+- The following variables exist and have value
+    - `provider-uri`
+    - `id-token-user-property`
+- The group `conjur/authn-oidc/<service-id>/users` exists and has `read` & `authenticate` 
+permissions on the webservice
+- The OIDC Provider (defined in the `provider-uri` variable) is responsive
+
+## Things we will not check
```

Comment:
@orenbm also the authenticators environment variable is not mentioned here, not in `things we will not check` and not in what we do plan to implement. Can you share why this was left out?

Reply:
It is mentioned in the general status check that the authenticator is enabled in the ENV. This check is shared across all authenticators. (https://github.com/cyberark/conjur/blob/authn-status-design/design/authenticators-status/authn_status_general.md#implementation-details)